tests/Fixer/Operator/BinaryOperatorSpacesFixerTest.php

```diff
@@ -556,6 +556,30 @@ function A(){}[$a] = $a[$c];
             null,
             ['operators' => ['|' => BinaryOperatorSpacesFixer::NO_SPACE]],
         ];
+
+        yield 'multiple shifts in single line (success)' => [
+            '<?php
+             $testA = $testB["abc"]    / $testC * 100;
+             $testD = $testE["abcdef"] / $testF * 100;
+            ',
+            '<?php
+             $testA = $testB["abc"]/$testC * 100;
+             $testD = $testE["abcdef"]/$testF * 100;
+            ',
+            ['default' => BinaryOperatorSpacesFixer::ALIGN_SINGLE_SPACE_MINIMAL],
+        ];
+
+        yield 'multiple shifts in single line (failed)' => [
```

Comment:
However, this case is failed because of this error below.

```
There was 1 failure:
1) PhpCsFixer\Tests\Fixer\Operator\BinaryOperatorSpacesFixerTest::testConfigured with data set "multiple shifts in single line (failed)" ('<?php\n             $testA = ...      ', '<?php\n             $testA = ...      ', array('align_single_space_minimal'))
Tokens collection built on expected code must not be marked as changed after fixing.
Failed asserting that true is false.

/PHP-CS-Fixer/tests/Test/AbstractFixerTestCase.php:519
/PHP-CS-Fixer/tests/Fixer/Operator/BinaryOperatorSpacesFixerTest.php:86
```

Reply:
The check that fails means that the expected code was changed.

If you replace the input snippet to `null` (lines 577-580 in the diff) and add `var_dump($code);` at the beginning of `PhpCsFixer\Tokenizer\Tokens::setCode`, the you will see that code was changing:
```cli
kub@:~/code/friendsofphp/PHP-CS-Fixer(master)$ ./vendor/bin/phpunit ./tests/Fixer/Operator/BinaryOperatorSpacesFixerTest.php --filter=failed
PHPUnit 9.6.15 by Sebastian Bergmann and contributors.

F                                                                                                                                               1 / 1 (100%)string(156) "<?php
             $testA = $testB["abc"]    / $testC * 10000000    * 100;
             $testD = $testE["abcdef"] / $testF * 10000000000 * 100;
            "
string(156) "<?php
             $testA = $testB["abc"]    / $testC * 10000000    * 100;
             $testD = $testE["abcdef"] / $testF * 10000000000 * 100;
            "
string(153) "<?php
             $testA = $testB["abc"]    / $testC * 10000000 * 100;
             $testD = $testE["abcdef"] / $testF * 10000000000 * 100;
            "
string(156) "<?php
             $testA = $testB["abc"]    / $testC * 10000000    * 100;
             $testD = $testE["abcdef"] / $testF * 10000000000 * 100;
            "
string(156) "<?php
             $testA = $testB["abc"]    / $testC * 10000000    * 100;
             $testD = $testE["abcdef"] / $testF * 10000000000 * 100;
            "


Time: 00:00.012, Memory: 8.00 MB

There was 1 failure:

1) PhpCsFixer\Tests\Fixer\Operator\BinaryOperatorSpacesFixerTest::testConfigured with data set "multiple shifts in single line (failed)" ('<?php\n
   $testA = ...      ', null, array('align_single_space_minimal'))
Tokens collection built on expected code must not be marked as changed after fixing.
Failed asserting that true is false.

/home/kuba/code/friendsofphp/PHP-CS-Fixer/tests/Test/AbstractFixerTestCase.php:569
/home/kuba/code/friendsofphp/PHP-CS-Fixer/tests/Fixer/Operator/BinaryOperatorSpacesFixerTest.php:86

FAILURES!
Tests: 1, Assertions: 3, Failures: 1.
```

Reply:
The problem here was that original `$tokens` was cloned and overridden (with `$tokens->setCode()`) for each `$toAlign` operator, even if code was not changed. I did some changes, so original `$tokens` is cloned only once, and content is overridden at the end only once too, but _only if cloned tokens' content was changed_. It should improve performance because collection is not re-created every time, but also it fixes problem with checking if expected code is changed after applying fixer.